src/article.py

```diff
@@ -23,35 +24,33 @@ def delete_prepositions(string, to_delete, case_sensitive=True):
     If case_sensitive is False, ignore if words in to_delete are
     captialized or not.
     """
-    # NOTE: it doesn't words in to_delete if they are followed e.g. by a
-    # comma. In practise, this is not a problem because preposition or articles
-    # are always surrounded by
+    # NOTE: it doesn't work for words in to_delete if they are followed
+    # e.g. by a comma. In our application, this is not a problem though.
     split_string = string.split()
     if not case_sensitive:
         to_delete_caps = [word.capitalize() for word in to_delete]
         to_delete += to_delete_caps
     for word in to_delete:
         while word in split_string:
             split_string.remove(word)
+
     return " ".join(split_string)
 
 
 def bib_title(string):
     """Helper function to create correct title for bibtex, i.e. curly braces
-    around capital words to actually print them in captial and escaping special
+    around capital words to actually print them in capital and escaping special
     characters.
     """
     caps = re.compile("[A-Z]")
     special = [r'"', r"{", r"}"]
     split_string = string.split()
     split_string = [escape_special_chars(w, special) for w in split_string]
-    split_string = [
-        "{{{}}}".format(w) if re.search(caps, w) else w for w in split_string
-    ]
+    split_string = [f"{{{w}}}" if re.search(caps, w) else w for w in split_string]
```

Comment:
This is quite complicated to follow. Can you provide an example as a comment of what it does? How about using `re.replace` rather than `search` with `if`?